src/parser/jobs/rpr/modules/OtherGauges.tsx

```diff
@@ -111,11 +116,7 @@ export class OtherGauges extends CoreGauge {
 
 		switch (action.id) {
 		case this.data.actions.ENSHROUD.id:
-			this.shroudGauge.spend(HIGH_SHROUD_MOD)
-			break
-
-		case this.data.actions.PLENTIFUL_HARVEST.id:
-			this.shroudGauge.generate(HIGH_SHROUD_MOD)
+			if (this.IsIdealHost === false) { this.shroudGauge.spend(HIGH_SHROUD_MOD) }
```

Comment:
this.actors.current.hasStatus should be enough, then you don't need to hook the application/removal etc.